tests/template_tests/syntax_tests/test_cycle.py

```diff
@@ -127,3 +127,39 @@ def test_cycle27(self):
     def test_cycle28(self):
         output = self.engine.render_to_string('cycle28', {'a': '<', 'b': '>'})
         self.assertEqual(output, '<&gt;')
+
+    @setup({
+        'cycle29': "{% cycle 'a' 'b' 'c' as cycler silent %}"
+                   "{% for x in values %}"
+                   "{% with doesnothing=irrelevant %}"
+                   "{% ifchanged x %}"
+                   "{% cycle cycler %}{{ cycler }}"
+                   "{% else %}"
+                   "{{ cycler }}"
+                   "{% endifchanged %}"
+                   "{% endwith %}"
+                   "{% endfor %}"})
+    def test_cycle29(self):
+        output = self.engine.render_to_string(
+            'cycle29', {
+                'irrelevant': 1,
+                'values': [1, 2, 3, 4, 5, 6, 7, 8, 8, 8, 9, 9, ]
+            })
+        self.assertEqual(output, 'bcabcabcccaa')
+
+    @setup({
+        'cycle30': "{% cycle 'a' 'b' 'c' as cycler silent %}"
+                   "{% for x in values %}"
+                   "{% ifchanged x %}"
+                   "{% cycle cycler %}{{ cycler }}"
+                   "{% else %}"
+                   "{{ cycler }}"
+                   "{% endifchanged %}"
+                   "{% endfor %}"})
+    def test_cycle30(self):
```

Comment:
Since this test already passes before the fix, it would be more clear to add it in a separate commit from the bug fix.